### PR TITLE
feat(create-article): CRUD for Article

### PR DESCRIPTION
**Pull request by fahdjamy**
#### What does this PR do?
- write an Article model
- create API endpoints for only authenticated users to create new articles.
- create API endpoints for all users to view available articles.
- create API endpoints for only authorized users to update or delete their own articles.
- write tests for the Article models
- write tests for the article API endpoints
#### Description of Task to be completed?
- Create API endpoint for users to retrieve single or all articles.
- Create API endpoint for users to create, delete and update an article
- It includes unit tests for the functionality
#### How should this be manually tested?
- Visit the PR's [app](https://ah-backend-summer-stagin-pr-17.herokuapp.com/api/v1/articles/) in the postman to view all articles or create a new article.
- Get all article
- Create an article
- To retrieve, update or delete single article Visit the PR's [app](https://ah-backend-summer-stagin-pr-17.herokuapp.com/api/v1/articles/<slug>) in the postman to view all articles or create a new article.
- Update an article
- Delete an article
- Retrieve an article.
#### Any background context you want to provide?
- On authors heaven, users should be able to create articles plus update or delete their own articles
#### What are the relevant pivotal tracker stories?
-  [#163383180](https://www.pivotaltracker.com/story/show/163383180)

### CODE REVIEW

**Review by karuhanga — APPROVED** [review]
LGTM.

**Comment by missvicki** [comment]
I think it would be nice if the Pull request started showing the output data as well. You could take some screenshots of how this is done on Postman. Again, this is just an idea. Otherwise, the work looks good.

**Review by oma0256 — APPROVED** [review]

File: authors/apps/articles/utils/model_helpers.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from ..models import Article
+
+def get_single_article_using_slug(slug):
+        # This method will be used to get a single article using a slug
+        # It will return a none is it tries getting an article with non exisiting slug
+        try:
+            obj = Article.objects.get(slug=slug)
+            return obj
+        except Article.DoesNotExist:
+            return None
```
Comment (codjoero) [inline authors/apps/articles/utils/model_helpers.py]: Hi @FahdJamy, great work man, you might want to take care of the End of Line after line 10

**Review by codjoero — APPROVED** [review]
Everything looks good, great job, good thing we went through the code earlier on.

File: authors/apps/articles/utils/model_helpers.py (hunk 1 above)
Comment (fahdjamy) [reply authors/apps/articles/utils/model_helpers.py]: Thanks. let me work on that now

**Review by pbkabali — APPROVED** [review]

**Pull request merged**